test/unit-test/cellular_pktio_utest.c

```diff
@@ -192,6 +191,7 @@ void setUp()
     tokenTableType = 0;
     recvCommFail = 0;
     setpktDataPrefixCBReturn = 0;
+    testInfiniteLoop = 0;
```

Comment:
do we still need testInfiniteLoop ?

Reply:
This flag is used to cover the following `else if` in opposite condition.
https://github.com/FreeRTOS/Lab-Project-FreeRTOS-Cellular-Library/blob/a8bd51d7c959f355d5f6ee20bc666aedcd59f190/common/src/cellular_pktio.c#L1037

Reply:
https://github.com/aws/amazon-freertos/blob/main/libraries/abstractions/transport/utest/transport_secure_sockets_utest.c#L274

Best practice should be using mock_somefile.h and control the return value with some_function_ExpectedAndReturn

Per-talk offline suggest to make this change in future refactor phases. Will not block this PR.